### PR TITLE
update doc theme and contributing guide

### PR DESCRIPTION
**Pull request by adehad**
# Summary of Changes
1. Move to a new theme that is being kept up to date and is more responsive, removing previous workarounds
2. Update dependencies following the addition of new doc deps
3. Update the contributing section

Reviewable via: https://jira--1474.org.readthedocs.build/

# TODO
- [x] Update to add details for the devcontainer

### CODE REVIEW

**Comment by adehad** [comment]
FYI @ssbarnea this proposes moving to `furo` from `sphinx-rtd-theme`

File: docs/contributing.rst [hunk 1]
```diff
@@ -37,17 +37,14 @@ Contributing Code
 Testing
 *******
 
-To test code run::
+Using tox::
```
Comment (adehad) [inline docs/contributing.rst:40]:
```suggestion
```

File: docs/jirashell.rst [hunk 2]
```diff
@@ -18,12 +24,17 @@ This is a specialized Python interpreter (built on IPython) that lets you explor
 Python code is acceptable input. The shell builds a ``JIRA`` client object for you (based on the launch parameters) and
 stores it in the ``jira`` object.
 
-Try getting an issue::
+Try getting an issue
+
+.. code-block:: bash
```
Comment (studioj) [inline docs/jirashell.rst:29]: Code block python

File: docs/jirashell.rst [hunk 3]
```diff
@@ -18,12 +24,17 @@ This is a specialized Python interpreter (built on IPython) that lets you explor
...
 
     In [1]: issue = jira.issue('JRA-1330')
 
 ``issue`` now contains a reference to an issue ``Resource``. To see the available properties and methods, hit the TAB
-key::
+key
+
+
+.. code-block:: bash
```
Comment (studioj) [inline docs/jirashell.rst:37]: Code block python

**Comment by ssbarnea** [comment]
I did not check it yet but I know and like furo. I would be against a non mainstream theme but furo is popular. Always good to pick something well maintained.

File: docs/jirashell.rst (hunk 2 above)
Comment (adehad) [reply docs/jirashell.rst:29]: For reference

Before:
image

After (looks the same, which is fine I guess):
image

File: docs/conf.py [hunk 4]
```diff
@@ -270,7 +265,7 @@
 napoleon_google_docstring = True
 napoleon_numpy_docstring = False  # Explicitly prefer Google style docstring
 napoleon_use_param = True  # for type hint support
-
+napoleon_use_rtype = False  # False so the return type is inline with the description.
```
Comment (adehad) [inline docs/conf.py:268]: So for reference this looks like

https://jira--1474.org.readthedocs.build/api.html#jira.client.JIRA.add_group


Before:
image

After:
image

**Pull request merged**